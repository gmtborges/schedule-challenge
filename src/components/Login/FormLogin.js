import React from 'react';
import { Form, FormField, Box, Button } from 'grommet';
import { Link } from 'react-router-dom';

const FormLogin = props => {
	return (
		<Form
			onSubmit={props.handleSubmit}
			messages={{ required: 'Campo obrigatório.' }}
		>
			<FormField
				name="email"
				label="E-mail"
				placeholder="seu@email.com"
				id="email"
				htmlFor="email"
				required
				validate={{
					regexp: /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/,
					message: 'E-mail inválido'
				}}
			/>
			<FormField
				type="password"
				name="password"
				id="password"
				htmlFor="password"
				label="Senha"
				required
			/>
			<Box direction="row" margin={{ top: '48px' }}>
				<Box width="180px">
					<Link to="/signup">
						<Button label="Criar conta" />
					</Link>
				</Box>
				<Box width="180px" margin={{ left: 'auto' }} alignSelf="end">
					<Button type="submit" label="Entrar" primary />
				</Box>
			</Box>
		</Form>
	);
};

export default FormLogin;
