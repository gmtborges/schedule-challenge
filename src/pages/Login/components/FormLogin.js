import React from 'react';
import { Form, FormField, Box, Button } from 'grommet';

const FormLogin = props => (
	<Form
		onSubmit={props.handleSubmit}
		messages={{ required: 'Campo obrigatório.' }}
	>
		<FormField
			name="email"
			label="E-mail"
			id="email"
			htmlFor="email"
			required
			validate={{
				regexp: /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/,
				message: 'E-mail inválido'
			}}
		/>
		<FormField
			type="password"
			name="password"
			id="password"
			htmlFor="password"
			label="Senha"
			required
		/>
		<Box direction="row" margin={{ top: '48px' }}>
			<Box width="180px">
				<Button type="submit" label="Criar conta" />
			</Box>
			<Box width="180px" margin={{ left: 'auto' }} alignSelf="end">
				<Button type="submit" label="Entrar" primary />
			</Box>
		</Box>
	</Form>
);

export default FormLogin;
