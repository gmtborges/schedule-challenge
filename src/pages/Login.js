import React, { Component } from 'react';
import { Box } from 'grommet';
import LeftSide from '../components/LeftSide';
import RightSide from '../components/RightSide';
import FormLogin from '../components/Login/FormLogin';

export default class Login extends Component {
	handleSubmit = event => {
		console.log(event.value);
		event.value = {};
	};

	render() {
		return (
			<Box direction="row" height="100vh">
				<LeftSide />
				<RightSide>
					<Box width="498px">
						<FormLogin handleSubmit={this.handleSubmit} />
					</Box>
				</RightSide>
			</Box>
		);
	}
}
