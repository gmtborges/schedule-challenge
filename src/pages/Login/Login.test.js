import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import Login from './Login';
import FormLogin from './components/FormLogin';

afterEach(cleanup);

test('should render Login page', () => {
	const { asFragment } = render(<Login />);

	expect(asFragment()).toMatchSnapshot();
});

test('should form contains email and password fields', () => {
	const { getByLabelText } = render(<Login />);
	const email = getByLabelText('E-mail');
	const password = getByLabelText('Senha');

	expect(email).toBeDefined();
	expect(password).toBeDefined();
});

test('should calls onSubmit with email and password filled', () => {
	const handleSubmit = jest.fn();
	const { getByLabelText, getByText } = render(
		<FormLogin handleSubmit={handleSubmit} />
	);
	const emailInput = getByLabelText('E-mail');
	const passwordInput = getByLabelText('Senha');
	fireEvent.change(emailInput, { target: { value: 'test@test.com' } });
	fireEvent.change(passwordInput, { target: { value: '123' } });
	getByText(/entrar/i).click();

	expect(handleSubmit).toHaveBeenCalledTimes(1);
});
