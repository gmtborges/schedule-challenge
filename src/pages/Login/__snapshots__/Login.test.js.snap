// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login page 1`] = `
<DocumentFragment>
  <div
    class="StyledBox-sc-13pk1d4-0 bygWGk"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 brnKMM"
    >
      <h1
        class="StyledHeading-sc-1rdh4aw-0 fNirbH"
      >
        Agendamento
      </h1>
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 fSQLwI"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 gZuWwm"
      >
        <form>
          <div
            class="FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              class="StyledText-sc-1sadyjn-0 bKLvLm"
              for="email"
            >
              E-mail
            </label>
            <div
              class="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 duvNfq"
                  for="email"
                  id="email"
                  label="E-mail"
                  name="email"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              class="StyledText-sc-1sadyjn-0 bKLvLm"
              for="password"
            >
              Senha
            </label>
            <div
              class="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 duvNfq"
                  for="password"
                  id="password"
                  label="Senha"
                  name="password"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="StyledBox-sc-13pk1d4-0 gLuuwC"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bwhsOh"
            >
              <button
                class="StyledButton-sc-323bzc-0 kHrfdt"
                type="submit"
              >
                Criar conta
              </button>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 hXvJrg"
            >
              <button
                class="StyledButton-sc-323bzc-0 aNlbA"
                type="submit"
              >
                Entrar
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
